### PR TITLE
Demo app

### PR DESCRIPTION
**Pull request by AliceYuan**
Initial demo app, shows json view.
Also incorporates integration scripts copied form https://github.com/SAAVY/magpie

screenshot 2015-07-16 10 46 59

screenshot 2015-07-16 10 47 54

### CODE REVIEW

File: web/app/setup.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from flask import url_for
+from flask_bootstrap import Bootstrap
+from flask_wtf import Form
+from flask_wtf.csrf import CsrfProtect
+from wtforms import SubmitField
+from wtforms import TextField
+from wtforms.validators import ValidationError
+
+
+api_url = "http://localhost:5000/website"
```
Comment (MeenuV) [inline web/app/setup.py]: Can we move the actual urls to a config file/constants file? Then we can assign api_url = dev, prod, etc.

File: web/app/setup.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class WebsiteForm(Form):
+
+    def is_valid_url(form, field):
+        if field.data == "":
+            error_msg = 'Must not be empty field'
+            flash("URL Field Error: " + error_msg, 'error')
+            raise ValidationError(error_msg)
+
+    text_field = TextField('Website URL:', [is_valid_url], description='Enter in a website url to parse, eg: http://www.youtube.com/watch?v=dQw4w9WgXcQ')
+    submit_button = SubmitField('Submit')
```
Comment (MeenuV) [inline web/app/setup.py]: Not necessary, but maybe call it `Get Metadata`?

File: web/app/setup.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        response = requests.get(api_url, params=args)
+        return response.json()
+
+    def make_pretty_json(response_json):
+        json_output = json.dumps(response_json, indent=4, separators=(',', ': '))
+        return json_output
+
+    def show_response_errors(response_json):
+        print response_json
+        if (response_json['status'] != 200):
```
Comment (MeenuV) [inline web/app/setup.py]: Extract 200 to constant?

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#! /bin/bash
+
+# ANSI COLOR CODES
+RED='\033[0;31m'
+YELLOW='\033[1;33m'
+BLUE='\033[34m'
+BOLD='\033[1m'
+NC='\033[0m' # No Color
+
+echo -e "${BOLD}================== BEGIN BUILD ====================${NC}"
```
Comment (MeenuV) [inline install.sh]: `CHECKING VIRTUAL ENV`

File: app/setup.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+from flask import Flask, render_template, flash
```
Comment (MeenuV) [inline app/setup.py]: Why is there an app/setup.py and web/app/setup.py? Contents seem kind of similar

File: web/install.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#! /bin/bash
+
+# ANSI COLOR CODES
+RED='\033[0;31m'
+YELLOW='\033[1;33m'
+BLUE='\033[34m'
+BOLD='\033[1m'
+NC='\033[0m' # No Color
+
+echo -e "${BOLD}================== BEGIN BUILD ====================${NC}"
```
Comment (MeenuV) [inline web/install.sh]: `CHECKING VIRTUAL ENV`

File: web/app/setup.py (hunk 2 above)
Comment (AliceYuan) [inline web/app/setup.py]: the class? this is just a form

Comment (MeenuV) [inline web/app/setup.py]: Oh no, I meant the "Submit" button.

File: app/setup.py (hunk 5 above)
Comment (AliceYuan) [inline app/setup.py]: my bad didn't want to commit this

**Comment by MeenuV** [comment]
:+1:

**Pull request merged**